flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/threadinfo/ThreadInfoSamplesRequest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.runtime.webmonitor.threadinfo;
+
+import org.apache.flink.api.common.time.Time;
+
+import javax.annotation.Nonnegative;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/** A wrapper for parameters of a thread info sampling request. */
+public class ThreadInfoSamplesRequest {
+    private final int requestId;
+    private final int numSubSamples;
+    private final Time delayBetweenSamples;
+    private final int maxStackTraceDepth;
+
+    /**
+     * @param requestId ID of the sampling request.
+     * @param numSamples The number of samples.
+     * @param delayBetweenSamples The time to wait between taking samples.
+     * @param maxStackTraceDepth The maximum depth of the returned stack traces.
+     */
+    public ThreadInfoSamplesRequest(
+            int requestId,
+            @Nonnegative int numSamples,
+            Time delayBetweenSamples,
+            int maxStackTraceDepth) {
```

Comment:
Is this allowed to be negative? `ThreadMXBean#getThreadInfo(long, int)` seems to reject negative depths at least

Reply:
Added `@Nonnegative` + check.